play.pokemonshowdown.com/src/battle-log.ts

```diff
@@ -353,20 +353,44 @@ export class BattleLog {
 			break;
 
 		case 'turn':
+			const goToTurnButton = document.createElement('button');
 			const h2elem = document.createElement('h2');
+			goToTurnButton.className = 'subtle';
 			h2elem.className = 'battle-history';
 			let turnMessage;
+			let turnNumber: number;
 			if (this.battleParser) {
 				turnMessage = this.battleParser.parseArgs(args, {}).trim();
 				if (!turnMessage.startsWith('==') || !turnMessage.endsWith('==')) {
 					throw new Error("Turn message must be a heading.");
 				}
 				turnMessage = turnMessage.slice(2, -2).trim();
+				turnNumber = parseInt(turnMessage.split(' ')[1].trim());
 				this.battleParser.curLineSection = 'break';
 			} else {
 				turnMessage = `Turn ${args[1]}`;
+				turnNumber = parseInt(args[1]);
 			}
-			h2elem.innerHTML = BattleLog.escapeHTML(turnMessage);
+			let userId: string | undefined = undefined;
+			if (window.app?.user?.get) {
+				userId = window.app.user.get('userid');
+			}
+			if (!this.scene?.battle.isReplay && userId && [
+				this.scene?.battle.p1.id,
+				this.scene?.battle.p2.id,
+				this.scene?.battle.p3?.id,
+				this.scene?.battle.p4?.id,
+			].includes(userId)) {
+				h2elem.innerHTML = BattleLog.escapeHTML(turnMessage);
+			} else {
+				goToTurnButton?.addEventListener?.('click', e => {
```

Comment:
```suggestion
				goToTurnButton.addEventListener('click', e => {
```

I'm pretty sure `goToTurnButton` is guaranteed to exist here, so the optional chaining feels unnecessary.